ipaserver/plugins/baseuser.py

```diff
@@ -728,3 +739,138 @@ def post_callback(self, ldap, dn, entry_attrs, *keys, **options):
         self.obj.convert_usercertificate_post(entry_attrs, **options)
 
         return dn
+
+
+class ModCertMapData(LDAPModAttribute):
+    attribute = 'ipacertmapdata'
+    takes_options = (
+        DNParam(
+            'issuer?',
+            cli_name='issuer',
+            label=_('Issuer'),
+            doc=_('Issuer of the certificate'),
+            flags=['virtual_attribute', 'no_create', 'no_update']
+        ),
+        DNParam(
+            'subject?',
+            cli_name='subject',
+            label=_('Subject'),
+            doc=_('Subject of the certificate'),
+            flags=['virtual_attribute', 'no_create', 'no_update']
+        ),
+        Bytes(
+            'usercertificate*', validate_certificate,
+            cli_name='certificate',
+            label=_('Certificate'),
+            doc=_('Base-64 encoded user certificate'),
+        ),
+    )
+
+    @staticmethod
+    def _build_mapdata(subject, issuer):
+        return u'X509:{issuer}{subject}'.format(
+            issuer='<I>{}'.format(issuer.x500_text()) if issuer else '',
+            subject='<S>{}'.format(subject.x500_text()) if subject else '',
+            )
```

Comment:
@HonzaCholasta 
I'm afraid I disagree here. AD supports both (see [Map a Certificate to a User Account](https://technet.microsoft.com/en-us/library/cc754866(v=ws.11).aspx)) and it is easy for us to provide a function that supports only issuer / only subject / issuer+subject, so why make it more difficult for the administrator? It is more user-friendly to use the issuer or subject options rather than specify the format X509:....